server/src/infra/ldap_handler.rs

```diff
@@ -527,6 +544,47 @@ impl<Backend: BackendHandler + LoginHandler + OpaqueHandler> LdapHandler<Backend
         Ok(vec![make_add_error(LdapResultCode::Success, String::new())])
     }
 
+    async fn do_compare(&mut self, request: LdapCompareRequest) -> LdapResult<Vec<LdapOp>> {
+        let req = make_search_request::<String>(
+            &self.ldap_info.base_dn_str,
+            LdapFilter::And(vec![LdapFilter::Equality(
+                "dn".to_string(),
+                request.dn.to_string(),
+            )]),
+            vec![],
+        );
+        let res = self.do_search_or_dse(&req).await?;
+        let entry = res.first().ok_or(LdapError {
+            code: LdapResultCode::OperationsError,
```

Comment:
Isn't that a NoSuchObject error?

Reply:
We always expect to have at least one element of type `SearchResultDone` because of [this](https://github.com/nitnelave/lldap/blob/main/server/src/infra/ldap_handler.rs#L439), right?

Reply:
Yes, there is always a `SearchResultDone`, you're right.

Reply:
Maybe we should update the message to say "Missing SearchDone message in search response"?